### ISSUE TITLE
MySQL SSL

### ISSUE THREAD

**Issue by OmarDarwish** [google/grr#643]
It would be awesome if we can tell GRR to connect to the specified MySQL instance using TLS. Something like `mysql`'s `--ssl-ca`, `--ssl-cert`, `--ssl-key` options.

**Comment by grrrrrrrrr** [comment]
Just to give an update here, we are going to add this soon. I have a first draft already implemented.

**Comment by dwdixon** [comment]
@grrrrrrrrr Thank you!  Any rough idea on if this might make it into the next release?  We're moving to a dedicated database server for GRR and are really weary of doing in prod so without TLS/SSL between GRR and it's backend mysql db.  Greatly appreciate your time/efforts on this.

**Comment by grrrrrrrrr** [comment]
https://github.com/google/grr/commit/0600c08275b396cedf003693d5931dc4d7a6c217 has the basic SSL support.

Feel free to try it out but note that the caveats @pidydx mentions apply.

**Comment by dwdixon** [comment]
Thank you @grrrrrrrrr, wondering if you are @pidydx could elaborate a bit on the degree of performance hit on GRR when enabling MySQL SSL and if you could also suggest a proper approach for using an ssh tunnel in a production setting with GRR<->MySQL?  

Separately, I was also wondering if we would be able to open an issue/feature request on the python mysql client library to support the model described?  Thank you much.

**Comment by pidydx** [comment]
I didn't do any measurements with GRR specifically.  I just found some DBA articles showing some testing comparisons between no encryption, ssh tunnels, ssl tunnels, and native ssl a few years ago.  In the past I have used autossh to set up the tunnels so the host running GRR sets up a tunnel to the mysql host and then you configure GRR to connect to localhost:3306 which gets sent through the tunnel to the mysql host.

I haven't dug into the library to know if the problem that makes that interface difficult isn't deeper in another library so I'm not sure how difficult it would be to change.
